core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -554,6 +555,32 @@ public String authority() {
     }
   }
 
+  private final class DirectFallbackExecutor implements Executor {
+    @Override
+    public void execute(final Runnable command) {
+      final boolean[] accepted = new boolean[1];
+      try {
+        ManagedChannelImpl.this.executor.execute(new Runnable() {
+          @Override
+          public void run() {
+            accepted[0] = true;
+            command.run();
+          }
+        });
+      } catch (RejectedExecutionException e) {
```

Comment:
I disagree. The rejection check continues to allow calls to `executor`. That's the race. `executor` may be provided by the application and it's not safe to call after termination. Or at least that's been my assumption.

I could believe the argument that we should always use the `executor`, even after termination, based on the theory that the application is expecting us to use `executor` (so it may be "magical") and it's the application's own fault for making an RPC after shutdown (not termination). But in that case we _mustn't_ use direct if there was a `RejectionExecutionException`. I'm not sure how many options we have, but we'd maybe need to propagate the exception to the caller and we never call their listener.

But this seems to be neither of these. I'm not sure what API theory this change is based on.